lib/gen_stage.ex

```diff
@@ -1815,22 +1867,31 @@ defmodule GenStage do
   def init({mod, args}) do
     case mod.init(args) do
       {:producer, state} ->
-        init_producer(mod, [], state)
+        init_producer(mod, [], state, nil)
 
       {:producer, state, opts} when is_list(opts) ->
-        init_producer(mod, opts, state)
+        init_producer(mod, opts, state, nil)
+
+      {:producer, state, opts, continue_or_hibernate} when is_list(opts) ->
```

Comment:
Not sure about the naming of `continue_or_hibernate` as it can also contain the `timeout`. Including that as well seems verbose, is there a shorter term for it?

Reply:
I'd call it `additional_info`.